samples/python/text_generation/react_sample.py

```diff
@@ -0,0 +1,200 @@
+        k = text.find("\nObservation:")
+        tool_name = text[i + len("\nAction:") : j].strip()
+        tool_args = text[j + len("\nAction Input:") : k].strip()
+        text = text[:k]
+    return tool_name, tool_args, text
+
+
+def call_tool(tool_name: str, tool_args: str) -> str:
+    if tool_name == "get_weather":
+        city_name = json5.loads(tool_args)["city_name"]
+        key_selection = {
+            "current_condition": [
+                "temp_C",
+                "FeelsLikeC",
+                "humidity",
+                "weatherDesc",
+                "observation_time",
+            ],
+        }
+        resp = requests.get(f"https://wttr.in/{city_name}?format=j1")
+        resp.raise_for_status()
+        resp = resp.json()
+        ret = {k: {_v: resp[k][0][_v] for _v in v} for k, v in key_selection.items()}
+        return str(ret)
+    elif tool_name == "generate_image":
+        tool_args = tool_args.replace("(", "").replace(")", "")
+        prompt = json5.loads(tool_args)["prompt"]
+        prompt = urllib.parse.quote(prompt)
+        return json.dumps(
+            {"image_url": f"https://image.pollinations.ai/prompt/{prompt}"},
+            ensure_ascii=False,
+        )
+    else:
+        raise NotImplementedError
+
+
+def llm_with_tool(llm_pipe, llm_config, prompt: str, history, list_of_tool_info=()):
+    chat_history = [(x["user"], x["bot"]) for x in history] + [(prompt, "")]
+    tokenizer = llm_pipe.get_tokenizer()
+    planning_prompt = build_input_text(tokenizer, chat_history, list_of_tool_info)
+
+    text = ""
+    while True:
+        # llm pipe output based planning_prompt and the text (previous output)
+        output = llm_pipe.generate(planning_prompt + text, llm_config, streamer)
+        # parse the output to get action
+        action, action_input, output = parse_latest_tool_call(output)
+        if action:
+            observation = call_tool(action, action_input)
+            observation_txt = f"\nObservation: = {observation}\nThought:"
+            print("\n\n- Getting information from the tool API -", observation_txt, "\n")
+            output += observation_txt
+            text += output
+        else:
+            text += output
+            break
+
+    new_history = []
+    new_history.extend(history)
+    new_history.append({"user": prompt, "bot": text})
+    return text, new_history
+
+
+def streamer(subword):
+    print(subword, end='', flush=True)
+    # Return flag corresponds whether generation should be stopped.
+    return openvino_genai.StreamingStatus.RUNNING
+
+ 
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('model_dir')
+    args = parser.parse_args()
+
+    device = 'CPU'  # GPU can be used as well
+    llm_model_path = args.model_dir
+
+    llm_pipe = openvino_genai.LLMPipeline(llm_model_path, device)
+    llm_config = openvino_genai.GenerationConfig()
+    llm_config.max_new_tokens = 256
```

Comment:
Follow up on https://github.com/openvinotoolkit/openvino.genai/pull/1926#discussion_r2037313273

The way to fix the error is to use model's config instead of creating it from scratch: `llm_config = llm_pipe.get_generation_config()`